README.md

```diff
@@ -31,82 +69,134 @@ Add this plugin like this:
-  console.log(subfeed)
-  // {
-  //   feedpurpose: 'mygame',
-  //   subfeed: '@___________________________________________=.ed25519',
-  //   keys: {
-  //     curve,
-  //     public,
-  //     private,
-  //     id
-  //   }
-  // }
-})
+  metadata: {}
 ```
 
-This has created a `keys` object for a new subfeed, which you can use to publish
-application-specific messages (such as for a game). The `details` argument
-always needs `feedpurpose` and `feedformat` (supports `classic` for ed25519
-normal SSB feeds, and `bendybutt-v1`).
+Meaning:
+- `metafeed` - the id of the feed this is underneath. As this is the topmost feed, this is empty
+- `feedpurpose` - a human readable ideally unique handle for this feed
+- `feedtype` - the feed format ("classic" or "bendybutt-v1" are current options)
+- `seed` - the data from which is use to derive the `keys` and `id` of this feed.
+- `keys` - cryptographic keys used for signing messages published by this feed (see [ssb-keys])
+- `id` - the feedId (same as `keys.id`)
+- `metadata` - additional data
 
-The subfeed is created only if it has not been found on your (given) `metafeed`.
-Notice that to find an existing matching subfeed, we pass the `visit` function
-on the 2nd argument. The `f` there is an object that describes the subfeed, with
-the fields:
+NOTES:
+- the `root` metafeed is unique - you have only one, and it has no metafeed (it's at the top!)
+- if you have a legacy `main` feed, this will also set that up as a subfeed of your `root` feed.
 
-- `feedformat`: the string `'classic'` or the string `'bendybutt-v1'`
-- `feedpurpose`: same string as used to create the subfeed
-- `subfeed`: the SSB identifier for this subfeed
-- `keys`: the [ssb-keys] compatible `{ curve, public, private, id }` object
-- `metadata`: the same object used when creating the subfeed
+### `sbot.metafeeds.findOrCreate(metafeed, find, details, cb)`
 
-## API
+Looks for the first subfeed of `metafeed` that satisfies the condition in `find`,
+or creates it matching the properties in `details`.
+
+This is strictly concerned with meta feeds and sub feeds that **you own**, not
+with those that belong to other peers.
+
+Arguments:
+- `metafeed` - the metafeed you are finding/ creating under, can be:
+    - *FeedDetail* object (as returned by `findOrCreate()` or `getRoot()`)
+    - *null* which is short-hand for the `rootFeed` (this will be created if doesn't exist)
+- `find` - method you use to find an existing *FeedDetail*, can be:
+    - *function* of shape `(FeedDetail) => boolean`
+    - *null* - this method will then return an arbitrary subfeed under provided `metafeed`
+- `details` - used to create a new subfeed if a match for an existing one is not found, can be 
+    - *Object*: 
+        - `details.feedpurpose` *String* any string to characterize the purpose of this new subfeed
+        - `details.feedformat` *String* either `'classic'` or `'bendybutt-v1'`
```

Comment:
:fire: yeah `feedformat` NOT `feedtype`

Reply:
oh yeah 
:fire: `id` instead of `subfeed` - the whole meta / sub thing is way too many concepts for beginners. here you can easily see `id === keys.id`. `FeedDetail.id` should be the unique identifier of the Feed we're considerring ... calling `FeedDatail.subfeed` is more like asking for this things subfeed. Very confusing

I was also tempted to change `metafeed` to `parentfeed` . More happy to leave this though

Reply:
I gave feedback about the naming `subfeed` when the meta feeds spec was being written, I think my suggestion was `id` or `feedId`. But it ended up staying as `subfeed`, and it's part of the spec, such that [go-metafeed](https://github.com/ssbc/go-metafeed) also uses `subfeed`. Not worth changing this name, IMO.

Reply:
Thanks for context. Will revert given its locked in already